### DISCUSSION TITLE
Generator use

### DISCUSSION THREAD

**Discussion by hansgeorgk** [Louisvdw/dbus-serialbattery#416] [Q&A] [answered]
Hi,

I have set the following values:

~~~
# Charge current control management referring to cell-voltage enable (True/False).
CCCM_CV_ENABLE = True
# Discharge current control management referring to cell-voltage enable (True/False).
DCCM_CV_ENABLE = True

# Set Steps to reduce battery current. The current will be changed linear between those steps
CELL_VOLTAGES_WHILE_CHARGING         = [3.55, 3.50, 3.45, 3.30]
MAX_CHARGE_CURRENT_CV                     = [   0,    2,  40,  90]

CELL_VOLTAGES_WHILE_DISCHARGING      = [2.70, 2.80, 2.90, 3.10]
MAX_DISCHARGE_CURRENT_CV                  = [   0,    5,  30,  200]
~~~

When I am charging with the Generator the Generator is going up and down according the CCL, which is adjusted based on the cell voltages, which are changing every second  - this is quite annoying ... :-(

I tried using this
`LINEAR_LIMITATION_ENABLE = True`
the steps are now a little smaller, but still the current requested from the Generator is still changing quite a lot.

Any recommended settings for Generator use?

cheers
HG

**Comment by Louisvdw** [comment, marked as answer]
I did plan to add a dampener in the code so that changes is applied more slowly. This would help for what you see here, but I think it would be best for with a generator to rather set (D)CCCM_CV_ENABLE to false and use a lower charge voltage.
So it's better to rather charge slower with the geni.
